### ISSUE TITLE
MDSelectionList set_selection_mode is automatically initiated when switching screen. 

### ISSUE THREAD

**Issue by gregynard** [kivymd/KivyMD#856]
### Description of the Bug

I don't know if this is a bug or there is something wrong with the way I made the program. My goal is to have two functionalities with the items on the MDSectionlist. The first one is when you pressed an item you will be redirected to a second screen and when you long pressed the item the selection is activated. Kindly see the below code for your reference. And please bare with me as I'm not as good as the other users. Thanks in advance. 

### Code and Logs

```python

from kivy.animation import Animation
from kivy.lang import Builder
from kivy.utils import get_color_from_hex
from kivymd.uix.menu import MDDropdownMenu, MDBoxLayout
from kivymd.app import MDApp
from kivymd.uix.behaviors import TouchBehavior
from kivymd.uix.list import TwoLineAvatarListItem, OneLineListItem
from kivymd.uix.card import MDCard, MDCardSwipe

KV = '''
MDBoxLayout:
    orientation: "vertical"
    
    MDToolbar:
        id: toolbar
        title: "Inbox"
        left_action_items: [["close", lambda x: app.call()]]
        right_action_items: [["magnify"], ["dots-vertical"]]
        md_bg_color: 0, 0, 0, 1

    MDBoxLayout:
        padding: "24dp", "8dp", 0, "8dp"
        adaptive_size: True
        MDLabel:
            text: "Today"
            adaptive_size: True
    ScreenManager: 
        id: sm
        Screen: 
            name: "main"
            ScrollView:
                MDSelectionList:
                    id: selection_list
                    spacing: "12dp"
                    overlay_color: app.overlay_color[:-1] + [.2]
                    icon_bg_color: app.overlay_color
                    on_selected: app.on_selected(*args)
                    on_unselected: app.on_unselected(*args)
                    on_selected_mode: app.set_selection_mode(*args)
        Screen
            name: "sub"
            MDLabel:
                text: "test"
        
    
'''

class Example(MDApp):
    overlay_color = get_color_from_hex("#6042e4")
    check = None
    items = ["test {}".format(x) for x in range(4)]

    def released(self, instance):
        if self.check:
            self.check = False
        else:
            if self.root.ids.selection_list.selected_mode:
               pass
            else:
                self.root.ids.sm.current = "sub"

    def call(self):
        self.root.ids.sm.current = "main"


    def build(self):
        return Builder.load_string(KV)

    def on_start(self):
        for item in self.items:
            x = OneLineListItem(text = item ,  _no_ripple_effect=True)
            self.root.ids.selection_list.add_widget(x)
            x.bind(on_release = self.released)



    def set_selection_mode(self, instance_selection_list, mode):
        if mode:
            md_bg_color = self.overlay_color
            left_action_items = [
                [
                    "close",
                    lambda x: self.root.ids.selection_list.unselected_all(),
                ]
            ]
            right_action_items = [["trash-can"], ["dots-vertical"]]
            Animation(md_bg_color=md_bg_color, d=0.2).start(self.root.ids.toolbar)
            self.root.ids.toolbar.left_action_items = left_action_items
            self.root.ids.toolbar.right_action_items = right_action_items

        else:
            md_bg_color = (0, 0, 0, 1)
            left_action_items = [["menu"]]
            right_action_items = [["magnify"], ["dots-vertical"]]
            self.root.ids.toolbar.title = "Inbox"
            Animation(md_bg_color=md_bg_color, d=0.2).start(self.root.ids.toolbar)
            self.root.ids.toolbar.left_action_items = left_action_items
            self.root.ids.toolbar.right_action_items = right_action_items


    def on_selected(self, instance_selection_list, instance_selection_item):
        self.root.ids.toolbar.title = str(
            len(instance_selection_list.get_selected_list_items())
        )
    def on_unselected(self, instance_selection_list, instance_selection_item):
        if instance_selection_list.get_selected_list_items():
            self.root.ids.toolbar.title = str(
                len(instance_selection_list.get_selected_list_items())
            )
        if int(len(instance_selection_list.get_selected_list_items())) < 1:
            self.check = True

Example().run()
```

### Screenshots

Add images to explain us this bug. Paste urls here.

Remove this section if no images here

### Versions

* OS: 
* Python: 
* Kivy: 
* KivyMD:

**Comment by HeaTTheatR** [comment]
@gregynard I'll check it out a little later ...

**Comment by gregynard** [comment]
@HeaTTheatR oki thanks!

**Comment by HeaTTheatR** [comment]
@gregynard 
```python
    def released(self, instance):
        def released(interval):
            if self.check:
                self.check = False
            else:
                if self.root.ids.selection_list.selected_mode:
                    pass
                else:
                    self.root.ids.sm.current = "sub"

        Clock.schedule_once(released, 0.2)
```

**Comment by gregynard** [comment]
Hi @HeaTTheatR thanks a lot.
